src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java

```diff
@@ -27,7 +30,7 @@ public Exception decode(String methodKey, Response response) {
 
         byte[] responseBody = null;
 
-        if (response.body() != null) {
+        if (response.body() != null && response.body().length() != null) {
```

Comment:
might consider response.body().length() > 0. It will not be null

Reply:
Tried with greater than 0 check initially but was throwing null pointer exception. So, added not null check.